### PR TITLE
Add Ongoing DAG/Update enddate for backlog DAG

### PR DESCRIPTION
**Pull request by sachitrajbhandari**
- Ongoing DAG added
- Update end date for backlog DAG to cover last processed date

### CODE REVIEW

File: dags/nci_c3_upload_s3.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+    args="-h dea-db.nci.org.au datacube -t -A -F,"
+    query="SELECT dsl.uri_body, ds.archived, ds.added, 
+    to_timestamp({{ next_execution_date.timestamp() }}) at time zone 'Australia/Canberra' as exec_dt 
+    FROM agdc.dataset ds 
+    INNER JOIN agdc.dataset_type dst ON ds.dataset_type_ref = dst.id 
+    INNER JOIN agdc.dataset_location dsl ON ds.id = dsl.dataset_ref 
+    WHERE dst.name='{{ params.product }}' 
+    AND (ds.added BETWEEN 
+    (to_timestamp({{ next_execution_date.timestamp() }}) at time zone 'Australia/Canberra' - interval '1 day')
```
Comment (omad) [inline dags/nci_c3_upload_s3.py:64]: I think that the `- interval '1 day'` should be removed, and the date check changed to be between `{{ execution_date.timestamp() }}` and `{{ next_execution_date.timestamp() }}`.

I'm sure this works, but it feels wrong to be arbitrarily specifying a day in two different places.

File: dags/nci_c3_upload_s3.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    # Be verbose and echo what we run
+    set -x
+
+    # Export AWS Access key/secret from Airflow connection module
+    export AWS_ACCESS_KEY_ID={{aws_creds.access_key}}
+    export AWS_SECRET_ACCESS_KEY={{aws_creds.secret_key}}
+
+    python3 '{{ work_dir }}/c3_to_s3_rolling.py' \
+            -f '{{ work_dir }}/{{ params.product }}.csv' \
+            -n '{{ params.nci_dir }}' \
```
Comment (omad) [inline dags/nci_c3_upload_s3.py:95]: If there a long versions of these arguments, I'd prefer to see them used here. No-one else is going to know what single letter arguments for a custom command are off the top of their head.

File: dags/nci_c3_upload_s3.py [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+}
+
+dag = DAG(
+    "nci_c3_upload_s3",
+    doc_md=__doc__,
+    default_args=default_args,
+    catchup=False,
+    schedule_interval="0 7 * * *",
+    max_active_runs=4,
+    default_view="graph",
```
Comment (omad) [inline dags/nci_c3_upload_s3.py:123]: `graph` was a nice default view when we weren't actually running things, but `tree` (the default) is a lot more helpful now that we're running things all the time.

File: dags/nci_c3_upload_s3_backlog.py [hunk 4]
```diff
@@ -31,8 +31,8 @@
 
 collection3_products = [
     ["ga_ls5t_ard_3", datetime(1986, 8, 15), datetime(2011, 11, 16)],
-    ["ga_ls7e_ard_3", datetime(1999, 5, 28), datetime(2020, 7, 31)],
-    ["ga_ls8c_ard_3", datetime(2013, 3, 19), datetime(2020, 7, 31)],
+    ["ga_ls7e_ard_3", datetime(1999, 5, 28), datetime(2020, 8, 5)],
+    ["ga_ls8c_ard_3", datetime(2013, 3, 19), datetime(2020, 8, 8)],
```
Comment (omad) [inline dags/nci_c3_upload_s3_backlog.py:35]: Don't change it, but it feels like a hack to have copied the entire dag and organise backlog running this way.

The Airflow CLI tool includes a `backfill` command, that could probably have been used for processing the backlog instead.

**Review by omad — APPROVED** [review]
Hi Sachit,

This looks okay, I've added a couple of comments for changes though. But feel free to merge this PR and make changes in a new one if you wish.

Cheers

**Comment by sachitrajbhandari** [comment]
> Hi Sachit,
> 
> This looks okay, I've added a couple of comments for changes though. But feel free to merge this PR and make changes in a new one if you wish.
> 
> Cheers

Hi Damien,
Thanks for the feedback. I've addressed your comments in this PR as they were reasonable. Review approval please!
Cheers,
Sachit.

**Review by omad — APPROVED** [review]
Thanks!

**Pull request merged**